v7/import_goodreads/import_goodreads.py

```diff
@@ -0,0 +1,134 @@
+from nikola import utils
+from nikola.utils import req_missing
+from nikola.plugins.basic_import import ImportMixin
+
+LOGGER = utils.get_logger('import_goodreads', utils.STDERR_HANDLER)
+
+class CommandImportGoodreads(Command, ImportMixin):
+    """Import a Goodreads RSS."""
+
+    name = "import_goodreads"
+    needs_config = False
+    doc_usage = "[options] rss_url"
+    doc_purpose = "import a Goodreads RSS"
+    cmd_options = [
+        {
+            'name': 'output_folder',
+            'long': 'output-folder',
+            'short': 'o',
+            'default': 'posts',
+            'help': 'Location to write imported content.'
+        },
+    ]
+
+    def _execute(self, options, args):
+        """Import Goodreads RSS."""
+
+        if feedparser is None:
+            req_missing(['feedparser'], 'import Goodreads RSS')
+            return
+
+        if not args:
+            print(self.help())
+            return
+
+        options['filename'] = args[0]
+        self.feed_export_file = options['filename']
+        self.output_folder = options['output_folder']
+        self.import_into_existing_site = True
+        channel = self.get_channel_from_file(self.feed_export_file)
+        self.import_posts(channel)
+
+    @classmethod
+    def get_channel_from_file(cls, filename):
+        return feedparser.parse(filename)
+
+    def import_posts(self, channel):
+        for item in channel.entries:
+            self.process_item(item)
+
+    def process_item(self, item):
+        if item.user_read_at:   # Only import books finished
+            self.import_item(item)
+
+    def import_item(self, item):
+        """Takes an item from the feed and creates a post file."""
+
+        link = item.link
+        if link.endswith('?utm_medium=api&utm_source=rss'):
+            link = link[:-30]
+
+        title = "Goodreads review: %s (%s)" % (item.title, item.author_name)
+
+        slug = utils.slugify(title)
+
+        # Needed becuase user_read_at can have a different locale
+        saved = locale.getlocale(locale.LC_ALL)
+        locale.setlocale(locale.LC_ALL, (None, None))
+        post_date = datetime.datetime.strptime(item.user_read_at[:-6], "%a, %d %b %Y %H:%M:%S")
+        locale.setlocale(locale.LC_ALL, saved)
+
+        content = ''
+        if item.get('user_review'):
+            content = item.get('user_review')
+
+        content += ("<br/><br/>" if content else "") + "Raiting: %s/5" % item.user_rating
+
+        content += "<br/><br/>Original: <a href=\"%s\">%s</a>" % (link, link)
+
+        tags = [item.author_name, item.title.replace(", ", " - "), "Goodreads review"]
+
+        content = self.transform_content(content)
+
+        self.write_metadata(
+            os.path.join(self.output_folder, slug + '.meta'),
+            title, slug, post_date, '', tags)
+        self.write_content(
+            os.path.join(self.output_folder, slug + '.html'),
+            content)
+
+    @staticmethod
```

Comment:
if you change post_date a few lines before, then you don't need this method at all